metrics/prometheus_test.go

```diff
@@ -24,3 +25,29 @@ func TestMultipleRegistrations(t *testing.T) {
 
 	p.Register("test", "counter")
 }
+
+func TestRaciness(t *testing.T) {
+	p := &PromMetrics{
+		Logger: &logger.MockLogger{},
+		Config: &config.MockConfig{},
+	}
+
+	err := p.Start()
+
+	assert.NoError(t, err)
+
+	p.Register("race", "counter")
+
+	// this loop modifying the metric registry and reading it to increment
+	// a counter should not trigger a race condition
+	for i := 0; i < 50; i++ {
+		go func(j int) {
+			metricName := fmt.Sprintf("metric%d", j)
+			p.Register(metricName, "counter")
+		}(i)
+
+		go func(j int) {
+			p.Increment("race")
+		}(i)
```

Comment:
Oh. Heh. A previous iteration of this used `p.Count("race", j)` which needed `i` to be passed in to trigger the _correct_ race condition. Doesn't need to be any more. 😶